utils/watermark_disk.go

```diff
@@ -0,0 +1,101 @@
+package utils
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+	"slices"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	bufferSize    = 512
+	numWatermarks = 10
+)
+
+type watermark struct {
+	position int64
+	data     []byte
+}
+
+func ApplyWatermarks(logicalName string) (func() error, error) {
+	// Write open
+	file, err := os.OpenFile(logicalName, os.O_WRONLY, 0)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	// Get disk or partition size
+	fileSize, err := file.Seek(0, io.SeekEnd)
+	if err != nil {
+		return nil, err
+	}
+	// Write watermarks on random locations
+	watermarks, err := writeWatermarks(file, fileSize, numWatermarks)
+	if err != nil {
+		return nil, err
+	}
+
+	checker := func() error {
+		file, err := os.OpenFile(logicalName, os.O_RDONLY, 0)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		for _, watermark := range watermarks {
+			_, err = file.Seek(watermark.position, io.SeekStart)
+			if err != nil {
+				return err
+			}
+			// Read the watermark written to the position
+			currentValue := make([]byte, bufferSize)
+			_, err = io.ReadFull(file, currentValue)
+			if err != nil {
+				return err
+			}
+			// Check if the watermark is still in the disk
+			if slices.Equal(currentValue, watermark.data) {
+				ErrorExistingWatermark := errors.New("Error existing watermark in the file")
+				return fmt.Errorf("%s | %w", logicalName, ErrorExistingWatermark)
```

Comment:
it would be good to know the position maybe?

Reply:
implemented and the tests